### PR TITLE
Add module defined prefix char check function

### PR DESCRIPTION
**Pull request by chinglee-iot**
Some module has different prefix format. For example, the sim7080 AT command response contains " " in their prefix.
```
+APP PDP: 0,ACTIVE
```

* Developer can override the CELLULAR_CHECK_IS_PREFIX_CHAR macro if different prefix char is allowed in their platform.

### CODE REVIEW

File: source/cellular_at_core.c [hunk 1]
```diff
@@ -43,10 +43,14 @@
 
 /*-----------------------------------------------------------*/
 
-#define CHECK_IS_PREFIX_CHAR( inputChar )                                 \
+#if ( CELLULAR_CUSTOM_CHECK_PFREFIX_CHAR == 0 )
```
Comment (andysun2015) [inline source/cellular_at_core.c:46]: Where's the default macro?

Comment (chinglee-iot) [reply source/cellular_at_core.c]: Fix the wrong macro name in the commit.

**Review by andysun2015 — DISMISSED** [review]

**Review by andysun2015 — APPROVED** [review]

File: source/cellular_at_core.c [hunk 2]
```diff
@@ -43,10 +43,14 @@
 
 /*-----------------------------------------------------------*/
 
-#define CHECK_IS_PREFIX_CHAR( inputChar )                                 \
+#if ( CELLULAR_CONFIG_MODULE_CHECK_PFREFIX_CHAR == 0 )
+    #define CHECK_IS_PREFIX_CHAR( inputChar )                             \
```
Comment (aggarg) [inline source/cellular_at_core.c:47]: Would adding a #ifndef be enough which will give a chance to override it in the config file? 

```
#ifndef CHECK_IS_PREFIX_CHAR
#define CHECK_IS_PREFIX_CHAR( inputChar ) \
( ( ( ( int32_t ) isalpha( ( ( int8_t ) ( inputChar ) ) ) ) == 0 ) && \
      ( ( ( int32_t ) isdigit( ( ( int8_t ) ( inputChar ) ) ) ) == 0 ) && \
      ( ( inputChar ) != '+' ) && ( ( inputChar ) != '_' ) )
#endif
```

Comment (chinglee-iot) [reply source/cellular_at_core.c]: Update in this PR.

**Review by andysun2015 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**